CMakeLists.txt

```diff
@@ -349,8 +360,11 @@ if (NOT STB_IMAGE_FOUND)
         SHOW_PROGRESS)
     list(GET STB_IMAGE_WRITE_DOWNLOAD_STATUS_LIST 0
         STB_IMAGE_WRITE_DOWNLOAD_STATUS)
+    check_hash (http://mlpack.org/files/stb/hash.md5 "${CMAKE_BINARY_DIR}/deps/${STB_DIR}"
+        HASH_CHECK_FAIL)
     if (STB_IMAGE_DOWNLOAD_STATUS EQUAL 0 AND
-        STB_IMAGE_WRITE_DOWNLOAD_STATUS EQUAL 0)
+        STB_IMAGE_WRITE_DOWNLOAD_STATUS EQUAL 0 AND
+        HASH_CHECK_FAIL EQUAL 0)
```

Comment:
Do you think we should provide a different error message if `HASH_CHECK_FAIL` is set?  Currently it will just print `STB_DOWNLOAD_STATUS` and `STB_DOWNLOAD_ERROR`, which would probably be confusing output if the md5sum failed.

Reply:
Added.

Reply:
Thanks for adding that, but we still get output like the following:

```
CMake Warning at CMake/CheckHash.cmake:23 (message):
  Could not verify md5 hash!
Call Stack (most recent call first):
  CMakeLists.txt:353 (check_hash)


CMake Warning at CMakeLists.txt:368 (message):
  Could not download stb! Error code : "No error"! Error log:


CMake Warning at CMakeLists.txt:370 (message):
  stb/stb_image.h is not installed.  Image utilities will not be available!
```

I think that we can avoid issuing the "could not download" errors if `HASH_CHECK_FAIL` is set.